app/lib/actions/katello/repository/fetch_pxe_files.rb

```diff
@@ -0,0 +1,46 @@
+module Actions
+  module Katello
+    module Repository
+      class FetchPxeFiles < Actions::EntryAction
+        middleware.use Actions::Middleware::KeepCurrentUser
+        middleware.use Actions::Middleware::ExecuteIfContentsChanged
+
+        input_format do
+          param :id, Integer
+          param :capsule_id, Integer
+        end
+
+        def run
+          repository = ::Katello::Repository.find(input[:id])
+          if repository.distribution_bootable? &&
+             repository.download_policy == ::Runcible::Models::YumImporter::DOWNLOAD_ON_DEMAND
+            capsule = nil
+            capsule = ::SmartProxy.find_by(:id => input[:capsule_id]) if input[:capsule_id]
+            os = Redhat.find_or_create_operating_system(repository)
+            Redhat::PXEFILES.keys.each do |pxe_file|
+              repo_path = repository.full_path(capsule).chomp("/")
```

Comment:
I'd probably say we should force the use of https on all repos, and use the uber cert to do so.

The reason i say this is because sometimes firewalls don't allow http connections (especially between a katello server and a smart proxy).